### PR TITLE
Give specific domain id for user/project

### PR DESCRIPTION
**Pull request by kajinamit**
Keystone supports "domain" and users, groups and projects belong to a specific domain, and it's possible that multiple entries with the same name exist in different domains.

This introduces the interface to specify the domain where the user or the project should exist, to avoid unexpected behavior caused by such "duplicate" entities.

### CODE REVIEW

File: modules/openstack/user.go [hunk 1]
```diff
@@ -76,8 +82,9 @@ func (o *OpenStack) CreateUser(
 func (o *OpenStack) GetUser(
 	log logr.Logger,
 	userName string,
+	domainID string,
```
Comment (bshephar) [inline modules/openstack/user.go:85]: Maybe if we need to change the function signature here anyway, we could pass in the User struct here? Which might improve the maintainability of this in the future if we need to add other parameters to it?

File: modules/openstack/user.go [hunk 2]
```diff
@@ -97,18 +106,20 @@ func (o *OpenStack) GetUser(
 func (o *OpenStack) DeleteUser(
 	log logr.Logger,
 	userName string,
+	domainID string,
```
Comment (bshephar) [inline modules/openstack/user.go:109]: Same here. Passing in a User struct might be more suitable moving forward.

File: modules/openstack/user.go (hunk 1 above)
Comment (kajinamit) [reply modules/openstack/user.go:85]: We can update the sigunature but that would eventually affect all of the functions we have in this openstack module, which can affect a few more operators. So I tend to leave it now but am open for opinion from the others.

Comment (stuggi) [reply modules/openstack/user.go:85]: yes, passing in a user struct would help to easy extend it in future without the immediate change of the consumers. if my grep was correct, only keystone, glance and heat use this module direct. but I guess we could also do it later

**Review by stuggi — APPROVED** [review]
/lgtm requires a change to at least keystone

**Comment by kajinamit** [comment]
I'll leave the suggested interface change because;

* We don't use the instance obtained by GetUser when calling DeleteUser so defining a struct might be redundant
* Changing the interface would need consistent update in the all resources
* I don't expect we need to update the interface later. We may need it when Keystone updates data structure but I assume it's very rare.

**Pull request merged**